### PR TITLE
feat: vite-node watch mode

### PR DESCRIPTION
**Pull request by poyoho**
fix: #889

don't close server when watch mode is open.And delete the moduleCache when the module refresh.

### CODE REVIEW

**Comment by lucas-jones** [comment]
When will this be release? Is there a way for me to use it now? 

Eager to try this out!

File: packages/vite-node/src/cli.ts [hunk 1]
```diff
@@ -93,5 +101,27 @@ async function run(options: CliOptions = {}) {
...
+  if (!options.watch)
+    await server.close()
+
+  event.on('updateModules', (modules: ModuleNode[]) => {
+    walkModules(modules, (module) => {
+      const { file, url } = module
+      const id = `/@fs/${slash(file!)}`
+      runner.moduleCache.delete(url)
+      runner.moduleCache.delete(id)
+      runner.executeId(id)
```
Comment (poyoho) [inline packages/vite-node/src/cli.ts:113]: * create
  just only `executeId`
* update
  delete the moduleCache and `executeId` to request vite transform module.
* delete
  delete module in moduleCache and request module. (TODO)

**Comment by poyoho** [comment]
> When will this be release? Is there a way for me to use it now?
> 
> Eager to try this out!

I think I need to improve the update process

File: packages/vite-node/src/cli.ts [hunk 2]
```diff
@@ -93,5 +93,16 @@ async function run(options: CliOptions = {}) {
...
-  await server.close()
+  if (!options.watch)
+    await server.close()
+
+  server.watcher.on('all', async(eventName, path) => {
+    // eslint-disable-next-line no-console
+    console.log(dim(`[update] ${path}`))
+    // because module don't had `import.meta.hot.accept`
+    // only can refresh all module
+    runner.moduleCache.clear()
```
Comment (antfu) [inline packages/vite-node/src/cli.ts:104]: I think at least we should keep the cache of node_modules.

**Comment by poyoho** [comment]
@antfu Should we add unit tests in the test directory?

**Pull request merged**